proposed/http-client/http-client.md

```diff
@@ -21,9 +21,9 @@ implementations. This would make libraries more stable since the reduced number
 dependencies and the likelihood to get in version conflicts is reduced.
 
 The second goal is that all HTTP clients should follow the [Liskov substitutions principle][Liskov].
-This means that all clients should act the same when sending a request. By default a HTTP client
-should not follow redirect nor throw exceptions on HTTP responses with status 4xx or 5xx.
-
+This means that all clients should act the same when sending a request. A consuming
+library SHOULD expect a HTTP client to NOT follow redirects NOR throw exceptions based
+on HTTP response statuses.
```

Comment:
Alternatively:

```php
const NO_FOLLOW = 1;
const CHECK_STATUS = 2;

public function sendRequest(RequestInterface $request, int $options = 0): ResponseInterface;
```

This way, consumers can elect for each use-case if they want to disable following, whether they want to map HTTP status-codes to exceptions, and possible other options added in the future.

I know that adding a bunch of options is a slippery road to start down, but both of these options seems to support common enough use-cases - consumer code will be a little more verbose in some cases, but it will be a lot more readable and easier to understand, since you don't need to know about configuration state being carried around by the client; and it respects LSP.